### ISSUE TITLE
[3.1] lighttable SIGSEGV _filemanager_zoom at dtgtk/thumbtable.c

### ISSUE THREAD

**Issue by Germano0** [darktable-org/darktable#4486]
darktable master.I was testing the new lighttable. I imported 1380 images, then darktable crashed.

[gdb trace.txt](https://github.com/darktable-org/darktable/files/4343403/gdb.txt)

**Platform (please complete the following information):**
- 3.1 master
 - OS: Fedora 31
 - OpenCL not active
 - VGA integrated in i5-2520M CPU

**Comment by AlicVB** [comment]
The backtrace is strange...
seems that you have changed the zoom level with the bottom slider from 21 to 6, right ? Do you remember if all the photos where loaded at this point ?
I've not reproduced atm, but I'll test with a bigger import.

**Comment by Germano0** [comment]
> seems that you have changed the zoom level with the bottom slider from 21 to 6, right ?

Yes

> Do you remember if all the photos where loaded at this point ?

Photos were not all loaded when it crashed. I started to experience this (reproducible) crash while I was testing why lighttable could not successfully import all the images

**Comment by AlicVB** [comment]
@Germano0 : can you test PR #4488 That should fix your issue.
I've not reproduce with your exact steps, but I've manage to reproduce the same crash at the same line with removing all thumbs from lib and then change zoom level.
So there's good chances that it'll fix your pb too.

**Comment by Germano0** [comment]
> @Germano0 : can you test PR #4488 That should fix your issue.

It fixes the crash but I have an empty lighttable despite having imported the images
![lighttable](https://user-images.githubusercontent.com/5477747/76873505-1ff40b00-686e-11ea-957e-bf3b9c654e03.jpg)

**Comment by Germano0** [comment]
To see them I am forced to go into "collected images" tab and select a film roll.

**Comment by AlicVB** [comment]
I've just added a commit in the PR that should fix that. Thanks for your tests !

**Comment by AlicVB** [comment]
too late... so it'll be for another PR !

**Comment by Germano0** [comment]
> too late... so it'll be for another PR !

Ok send me the URL so I can test it. If you don't want to append other messages here, you can find my e-mail address at https://src.fedoraproject.org/rpms/darktable/blob/master/f/darktable.spec#_232

**Comment by AlicVB** [comment]
@Germano0 : PR #4491 :)

**Comment by Germano0** [comment]
> @Germano0 : PR #4491 :)

darktable shows only 283 images over the 1387 imported. It's the same bug I was trying to reproduce when I experienced the crash that you just fixed

**Comment by AlicVB** [comment]
hmm... you speak once the loading job is finished ?
and if you go to darkroom and back, or reset the collect module, does it show them all ?

**Comment by Germano0** [comment]
> hmm... you speak once the loading job is finished ?

yes

>  and if you go to darkroom and back, or reset the collect module, does it show them all ?

To better reproduce the problem I:

1. erased ~/.config/darktable
2. opened darktable, increased various memory options in lighttable, then closed and reopened darktable
3. imported the folder that contained various subfolders, by enabling "import folders recursively". After the import has been completed, only 283 images were shown over 1387 imported.
4. closed darktable
5. reopened darktable. Without touching anything, darktable by itself started to show all 1387 images in the lighttable as it should have done before at point 3

**Comment by AlicVB** [comment]
right now I can't reproduce...
Some more questions : 
- does it reproduce with another folder with different images
- can you show me a screenshot of dt when you have "just" 283 images (to see scrollbars, etc...)
- are the 283 images the firsts of the collection or some random images ?
- if you change the zoom level *before* the import, does the 283 number change ?

Thanks (and sorry to annoy you with all my tests...)

**Comment by Germano0** [comment]
>     * does it reproduce with another folder with different images

While trying, I triggered another unexpected behaviour. See https://youtu.be/dxn0lIuLMXY
By the way, even if I could not see any image, I managed to click on "select all images"->"remove"
Then I tried again and I managed to see some images, but still an incomplete set of other imported images. Closing darktable and reopening it again, triggered lighttable showing all of the images. So yes I can reproduce the problem with other folders

> can you show me a screenshot of dt when you have "just" 283 images (to see scrollbars, etc...)

![darktable_lighttable](https://user-images.githubusercontent.com/5477747/76909835-9fa0ca80-68ac-11ea-93c2-26c448882188.jpg)


> are the 283 images the firsts of the collection or some random images ?

283 images were not the firsts of a collection, I think are just random images

> if you change the zoom level _before_ the import, does the 283 number change ?

No.

**Comment by AlicVB** [comment]
Ok, so dt load all your images, there's "just" a drawing problem.
Can you test that branch : https://github.com/AlicVB/darktable/tree/lt_fix_import ? I've added a full refreshing at the end of the import.
Thanks

**Comment by Germano0** [comment]
>  Can you test that branch : https://github.com/AlicVB/darktable/tree/lt_fix_import ? I've added a full refreshing at the end of the import.

All 1387 images imported and shown correctly.
"Select all"->"Remove" works fine too

**Comment by AlicVB** [comment]
great ! I've just done the PR. Thanks again for your help in debugging
